bool us_on
---
bool was_set

